### PR TITLE
Add sumo dependency to setup.py instead of using utility

### PR DESCRIPTION
**Pull request by qianyi-sun**
linked issue: #1299

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,97 +1,98 @@
...
+    include_package_data=True,
+    zip_safe=True,
+    python_requires=">=3.7",
+    install_requires=[
+        # setuptools:
+        #   tensorboard needs >=41
+        #   50.0 is broken: https://github.com/pypa/setupatools/issues/2353
+        "setuptools>=41.0.0,!=50.0",
+        "cached-property>=1.5.2",
+        "eclipse-sumo>=1.10.0",
```
Comment (Gamenot) [inline setup.py:27]: This here is `sumo`.

File: setup.py [hunk 2]
```diff
@@ -1,97 +1,98 @@
...
-        "waymo": ["waymo-open-dataset-tf-2-2-0"],
-    },
-    entry_points={"console_scripts": ["scl=cli.cli:scl"]},
-)
+from os import path
+
+from setuptools import find_packages, setup
+
+this_dir = path.abspath(path.dirname(__file__))
+with open(
```
Comment (Gamenot) [inline setup.py:6]: nit: The whitespace change is painful, I would prefer if the whitespace is reverted. :face_with_head_bandage:

**Review by Gamenot — APPROVED** [review]

**Comment by Gamenot** [comment]
I am slightly curious as to why [test_sumo_map](https://github.com/huawei-noah/SMARTS/runs/5366872656?check_suite_focus=true) is failing.

```
        route = routes[0]
        db = route.distance_between(point, (198, 65.20, 0))
>       assert db == 134.01
E       assert 134.0 == 134.01
E         +134.0
E         -134.01
```

File: setup.py (hunk 2 above)
Comment (sah-huawei) [reply setup.py:6]: agreed!!

File: tests/test_setup.py [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+    sys.exit(-1)
+
+
+# Check SUMO_HOME is properly set
+SUMO_HOME = "SUMO_HOME"
+sumo_path = os.getenv(SUMO_HOME)
+if sumo_path is None:
+    exit_with_error("{} environment variable is not set".format(SUMO_HOME))
+elif not os.path.exists(sumo_path):
+    exit_with_error("{} path: {} is invalid".format(SUMO_HOME, sumo_path))
```
Comment (sah-huawei) [inline tests/test_setup.py:19]: Is this just another whitespace change?  (I don't see any other diffs.)

File: README.md [hunk 4]
```diff
@@ -60,10 +60,6 @@ cd <path/to/SMARTS>
 # Install the system requirements, then follow the instructions for setting up the SUMO_HOME environment variable. You may use the `-y` option to enable automatic assumption of "yes" to all prompts to avoid timeout from waiting for user input.
```
Comment (sah-huawei) [inline README.md:60]: I think this comment needs to be updated too.

**Review by sah-huawei — CHANGES_REQUESTED** [review]
You'll need to update the documentation files in the `doc` folder too.  (I think at least both `setup.rst` and `SUMO_TROUBLESHOOTING.md` need to be updated there.  Maybe more.)

File: tests/test_setup.py (hunk 3 above)
Comment (qianyi-sun) [reply tests/test_setup.py:19]: yes, it is

File: README.md (hunk 4 above)
Comment (qianyi-sun) [reply README.md:60]: yes, and after we eventually move sumo to optional, we might need to modify this README.md again.

File: setup.py [hunk 5]
```diff
@@ -24,8 +24,8 @@
         #   50.0 is broken: https://github.com/pypa/setupatools/issues/2353
         "setuptools>=41.0.0,!=50.0",
         "cached-property>=1.5.2",
-        "eclipse-sumo>=1.10.0",
         "click>=8.0.3",  # used in scl
+        "eclipse-sumo>=1.11.0",  # sumo
```
Comment (Adaickalavan) [inline setup.py:28]: Using `eclipse-sumo==1.11.0` previously threw an error message. See https://github.com/huawei-noah/SMARTS/issues/1155

Have we checked whether this issue is addressed? If not, I would suggest to use `eclipse-sumo==1.10.0` .

Comment (qianyi-sun) [reply setup.py:28]: actually the newest version of sumo is 1.12.0, given that issue 1155, maybe we shouldn't try 1.12.0 too?

**Review by Adaickalavan — APPROVED** [review]

**Comment by Adaickalavan** [comment]
> I am slightly curious as to why [test_sumo_map](https://github.com/huawei-noah/SMARTS/runs/5366872656?check_suite_focus=true) is failing. Maybe it is due to `sumo` version `1.12.0`?
> 
> ```
>         route = routes[0]
>         db = route.distance_between(point, (198, 65.20, 0))
> >       assert db == 134.01
> E       assert 134.0 == 134.01
> E         +134.0
> E         -134.01
> ```

CI tests appear to pass after using `eclipse-sumo==1.10.0`. So it seems, both `eclipse-sumo==1.11.0` and `eclipse-sumo==1.12.0` might have unaddressed issues.

**Review by sah-huawei — APPROVED** [review]

**Pull request merged**